python/desc/lsst_camera_readout/lsst_camera_readout.py

```diff
@@ -60,13 +69,26 @@ class ImageSource(object):
         Object containing the readout properties of the sensors in the
         focal plane, extracted from the segmentation.txt file.
     '''
-    def __init__(self, eimage_file, seg_file=None, sensor_id=None,
-                 add_read_noise=True):
+    def __init__(self, eimage_file=None, image_array=None, seg_file=None, 
```

Comment:
Setting defaults for both `eimage_file` and `image_array` makes it impossible for syntax checkers like `pylint` to catch cases where an `ImageSource` object is created without any arguments, i.e., `foo = ImageSource()` would be identified as syntactically correct, though it will fail when run.

I think it would be better to make the constructor take an numpy natively, and create a factory method to generate the object from an eimage file.

The signatures of these two functions would look like
```
    def __init__(self, image_array, exptime, sensor_id, seg_file=None, add_read_noise=True):


    @staticmethod
    def create(eimage_file, seg_file=None, sensor_id=None, add_read_noise=True):
```
This way the client code is required to provide `image_array, exptime, sensor_id` to the constructor, the `create(...)` function will require an `eimage_file` argument, and `pylint` will be able to check both cases reliably.

I think it will just be easiest for me to implement these changes directly in master.